src/main/java/org/folio/processing/matching/matcher/Matcher.java

```diff
@@ -30,6 +32,12 @@
     MatchDetail matchDetail = matchProfile.getMatchDetails().get(0);
 
     Value value = matchValueReader.read(eventPayload, matchDetail);
+    if (value != null && value.getType().equals(Value.ValueType.STRING)) {
+      value = MatchIdProcessorUtil.retrieveIdFromContext(matchDetail.getExistingMatchExpression().getFields().get(0).getValue(),
```

Comment:
what if entity should be matched by the String value itself?

Reply:
@KaterynaSenchenko good question! 
In the case, where the field should be matched by String simple value (not via ID retrieve logic), the Value here will be the same as we pass to the "retrieveIdFromContext" method.
So, if code can`t find the key from Map for relations then it will return the same value. 
See this code from this method:

`    JsonArray jsonArray = matchingParams.getJsonArray(relation);
    if (jsonArray == null) {
      return value;
    }`